File: 1. The LeetCode Beginner's Guide/0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* middleNode(struct ListNode* head) {
    struct ListNode* one_step = head;
    struct ListNode* two_step = head;
    while ((two_step != NULL) && (two_step->next != NULL)) {
        two_step = two_step->next->next;
        one_step = one_step->next;
    }
    return one_step;
}

File: A. Easy/0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if (head.next == None):
            return head
        
        count = 1
        node_list = list()
        node_list.append(head)

        temp = head
        while (temp.next != None):
            count += 1
            node_list.append(temp.next)
            temp = temp.next
        
        node_num = (count // 2)
        return node_list[node_num]